exporter/src/main/as/flump/export/AtlasImpl.as

```diff
@@ -0,0 +1,136 @@
+        _height = h;
+        _borderSize = borderSize;
+        _mask = new BitmapData(_width, _height, true, 0);
+        _mask.lock();
+        _scaleFactor = scaleFactor;
+        _quality = quality;
+    }
+
+    public function get area () :int { return _width * _height; }
+
+    public function get scaleFactor () :int { return _scaleFactor; }
+
+    public function get quailty () :String { return _quality; }
+
+    public function get filename () :String { return name + AtlasMold.scaleFactorSuffix(_scaleFactor) + ".png"; }
+
+    public function get used () :int {
+        var used :int = 0;
+        _nodes.forEach(function (n :Node, ..._) :void {
+            used += n.paddedBounds.width * n.paddedBounds.height;
+        });
+        return used;
+    }
+
+    public function toMold () :AtlasMold {
+        const mold :AtlasMold = new AtlasMold();
+        mold.file = this.filename;
+        _nodes.forEach(function (node :Node, ..._) :void {
+            const tex :SwfTexture = node.texture;
+            const texMold :AtlasTextureMold = new AtlasTextureMold();
+            texMold.symbol = tex.symbol;
+            texMold.bounds = new Rectangle(node.bounds.x, node.bounds.y, tex.w, tex.h);
+            texMold.origin = new Point(tex.origin.x, tex.origin.y);
+            mold.textures.push(texMold);
+        });
+        return mold;
+    }
+
+    public function toBitmap () :BitmapData {
+        if (_bitmapData == null) {
+            var constructed :Sprite = new Sprite();
+            var collapsedBounds :Rectangle = new Rectangle();
+            _nodes.forEach(function (node :Node, ..._) :void {
+                const bm :Bitmap = new Bitmap(node.texture.toBitmapData(_borderSize), "auto", true);
+                constructed.addChild(bm);
+                bm.x = node.paddedBounds.x;
+                bm.y = node.paddedBounds.y;
+                collapsedBounds = collapsedBounds.union(node.paddedBounds);
+            });
+            _bitmapData = Util.renderToBitmapData(constructed,
+                    Util.nextPowerOfTwo(collapsedBounds.x + collapsedBounds.width),
+                    Util.nextPowerOfTwo(collapsedBounds.y + collapsedBounds.height),
+                    quailty);
+        }
+        return _bitmapData;
+    }
+
+    // Try to place a texture in this atlas
+    public function place (tex :SwfTexture, xx :uint, yy :uint) :void {
+        var w :int = tex.w + (_borderSize * 2);
+        var h :int = tex.h + (_borderSize * 2);
+        if (w > _width || h > _height) {
+            throw new Error("Tried to place a texture outside of the atlas. This is a bug.");
+        }
+        if (isMasked(xx, yy, w, h)) {
+            throw new Error("Tried to place a texture over another texture. This is a bug.");
+        }
+        var node :Node = new Node(xx, yy, _borderSize, tex);
+        _nodes.push(node);
+        setMasked(node.paddedBounds.x, node.paddedBounds.y, node.paddedBounds.width, node.paddedBounds.height);
+    }
```

Comment:
in the old logic this was used to check whether the texture fits into the atlas. Now it just does a double check